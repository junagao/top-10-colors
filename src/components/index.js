export { default as ColorList } from 'components/ColorList'
export { default as ColorItem } from 'components/ColorItem'
export { default as Header } from 'components/Header'
export { default as Rating } from 'components/Rating'
export { default as Star } from 'components/Star'
