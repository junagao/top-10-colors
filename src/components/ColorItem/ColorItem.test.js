import React from 'react'
import { shallow } from 'enzyme'
import { ColorItem, Rating } from 'components'

describe('ColorItem', () => {
  let wrapper

  const mockColor = {
    id: 'fc352593-c91d-4cf6-b689-7478ee2b61c8',
    name: 'orange',
    hex: '#FFA500',
    rgb: 'rgb(255, 165, 0)',
    rating: 3,
  }

  beforeEach(() => {
    wrapper = shallow(
      <ColorItem
        id={mockColor.id}
        name={mockColor.name}
        hex={mockColor.hex}
        rgb={mockColor.rgb}
        rating={mockColor.rating}
        onRate={jest.fn()}
      />,
    )
  })

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render one Card element', () => {
    expect(wrapper.find('Card').length).toEqual(1)
  })

  it('should render one Preview element with prop `hex`', () => {
    expect(wrapper.find('Preview').length).toEqual(1)
    expect(wrapper.find('Preview').prop('hex')).toBeTruthy()
    expect(wrapper.find('Preview').prop('hex')).toEqual('#FFA500')
  })

  it('should render one Title element with the color name as title', () => {
    expect(wrapper.find('Title').length).toEqual(1)
    expect(wrapper.find('Title').children().text()).toEqual('orange')
  })

  it('should render two Text elements with the color hex and rgb respectively', () => {
    expect(wrapper.find('Text').length).toEqual(2)
    expect(wrapper.find('Text').children().at(0).text()).toEqual('#FFA500')
    expect(wrapper.find('Text').children().at(1).text()).toEqual(
      'rgb(255, 165, 0)',
    )
  })

  it('should render one instance of Rating component', () => {
    expect(wrapper.find(Rating).length).toEqual(1)
  })
})
