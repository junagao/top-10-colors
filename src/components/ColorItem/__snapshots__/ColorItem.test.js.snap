// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorItem should render correctly given the required props 1`] = `
<Card>
  <Preview
    hex="#FFA500"
  />
  <Title>
    orange
  </Title>
  <Text>
    #FFA500
  </Text>
  <Text>
    rgb(255, 165, 0)
  </Text>
  <Rating
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    onRate={[MockFunction]}
    rating={3}
    totalStars={5}
  />
</Card>
`;
